app/models/percentiles_by_calendar_date.rb

```diff
@@ -0,0 +1,7 @@
+class PercentilesByCalendarDate < ActiveRecord::Base
+  include PercentileSerialisation
+
+  def self.refresh
+    Scenic.database.refresh_materialized_view(table_name, concurrently: false)
```

Comment:
It requires a unique index for each table, which is easy enough to add. The only downside to not do it concurrently is it locks the table for ```select```, which in our case is not a problem at as the view will be refresh once a day at midnight. So i didn't bother with concurrent refresh.